nvtabular/workflow.py

```diff
@@ -477,10 +477,10 @@ def get_final_cols_names(self, col_type):
         -----------
         col_type : str
         """
-        col_names = []
+        col_names = set()
         for c_names in self.columns_ctx[col_type].values():
-            col_names.extend(c_names)
-        return col_names
+            col_names.add(c_names)
+        return list(col_names)
```

Comment:
I don't think this is correct - have you tested this? 

I think you will get an error saying 'unhashable type list' or similar when running.  ('add' on a set object isn't  the same operation as 'extend' on a list - you probably meant for ```col_names.update(c_names)``` instead).

Also can you add a unittest for this function (and potentially even the whole writing ctr workflow)?

Reply:
Sorry. My bad, I changed from list to set and I didn't expect that problem. I will be using update.

I am adding unittest.

Reply:
This is that I have done so far for [unit test](https://github.com/NVIDIA/NVTabular/blob/c96f8dfdd5351203fdddf3a7afe89f273901e7d5/tests/unit/test_io.py#L91).

I don't know how to deal exactly with reading/writing files. I should be checking that files have been written correctly to ensure correct execution.

I am getting a deadlock when running this test that I assume it is related with that, because I have been running a program successfully.

I haven't seen any similar to follow the same format, but I may be missing something.